modules/btcpay/src/Invoice/Processor.php

```diff
@@ -243,41 +250,84 @@ public function paymentReceivedCreateAfter(BitcoinPayment $bitcoinPayment)
 		// Grab the secure key
 		$secureKey = $invoiceData['metadata']['posData'];
 
-		// Generate an order only if there is not another one with this cart
-		if ($existingOrder = \Order::getByCartId($bitcoinPayment->getCartId())) {
-			$message = \sprintf('[INFO] Invoice %s already has an order %s', $bitcoinPayment->getInvoiceId(), $existingOrder->reference);
-			\PrestaShopLogger::addLog($message, \PrestaShopLogger::LOG_SEVERITY_LEVEL_INFORMATIVE, null, 'BitcoinPayment', $bitcoinPayment->getId());
-
-			return;
-		}
+		\PrestaShopLogger::addLog(\sprintf("[INFO] Creating actual order for invoice '%s'", $invoice->getId()), \PrestaShopLogger::LOG_SEVERITY_LEVEL_INFORMATIVE, null, 'BitcoinPayment', $bitcoinPayment->getId());
 
 		$this->module->validateOrder(
 			$bitcoinPayment->getCartId(),
 			$orderStatus,
 			$bitcoinPayment->getAmount(),
-			$this->module->displayName, // BTCPay
-			null, //message should be new Message
-			[], // extra variables for mail
-			null, //currency special
-			false, // don't touch amount
-			$secureKey
+			$this->module->displayName,
+			null,
+			[],
+			null,
+			false,
+			$secureKey,
+			$this->context->shop
 		);
 
 		// Get the new order ID
 		$order = \Order::getByCartId($bitcoinPayment->getCartId());
 
+		// Store the new order ID and set the proper status
 		$bitcoinPayment->setOrderId($order->id);
 		$bitcoinPayment->setStatus($orderStatus);
 
-		// Add a message if the user paid too late
-		if ($invoice->isPaidLate()) {
-			\PrestaShopLogger::addLog('[INFO] User paid after expiration for this invoice', \PrestaShopLogger::LOG_SEVERITY_LEVEL_INFORMATIVE, null, 'Order', $order->id);
+		// Update the object
+		if (false === $bitcoinPayment->update(true)) {
+			$error = \sprintf('[ERROR] Could not update bitcoin_payment: %s', \Db::getInstance()->getMsgError());
+			\PrestaShopLogger::addLog($error, \PrestaShopLogger::LOG_SEVERITY_LEVEL_ERROR, null, 'BitcoinPayment', $bitcoinPayment->getId());
+
+			throw new \RuntimeException($error);
 		}
+	}
 
-		// Add a message if the overpaid
-		if ($invoice->isOverpaid()) {
-			\PrestaShopLogger::addLog('[INFO] User overpaid for this order', \PrestaShopLogger::LOG_SEVERITY_LEVEL_INFORMATIVE, null, 'Order', $order->id);
+	/**
+	 * @throws \JsonException
+	 * @throws \PrestaShopDatabaseException
+	 * @throws \PrestaShopException
+	 */
+	public function paymentSettled(BitcoinPayment $bitcoinPayment): void
```

Comment:
Something here doesn't go as expected when dealing with non USD payments.

![image](https://github.com/btcpayserver/prestashop-plugin/assets/27309467/3b8ce7ea-086d-4512-bf59-3ee48c535a4b)

Reply:
When the order changes to a state where the order state is `paid`, Prestashop will create an invoice no matter what. During one of the refactors I disabled the option that it should use existing payments (thus creating a new one). 

This has been resolved now.